index.js

```diff
@@ -158,27 +192,39 @@ const line = []
 // console.log(takeATicketNumber(line))
 // // => `Welcome. You are ticket number 3`
 
-// console.log("%c----------", "color: red") 
+console.log("%c----------", "color: red") 
 
 // ***** Scope & Closures - Question 2 *****
 // *** Uncomment the lines below to test
-// console.log("%cScope & Closures - Question 2", "color: red")
+console.log("%cScope & Closures - Question 2", "color: red")
+function ticketNumberGeneratorFunc(){
+  let num = 0
+
+  function takeATicketNumber(line) {
+    num ++
+    line.push(num)
+  
+    return `Welcome. You are ticket number ${num}`
+  }
 
-// const newLine = []
-// const takeATicketNumberFunc = ticketNumberGeneratorFunc()
-// console.log(takeATicketNumberFunc(newLine))
-// // => `Welcome. You are ticket number 1`
+  return takeATicketNumber
 
-// console.log(takeATicketNumberFunc(newLine))
-// // => `Welcome. You are ticket number 2`
+}
+const newLine = []
+const takeATicketNumberFunc = ticketNumberGeneratorFunc()
+console.log(takeATicketNumberFunc(newLine))
+// => `Welcome. You are ticket number 1`
```

Comment:
Great job using a closure to solve this problem. Your pull request says that you did not do the Bonus. But this closure looks good to me. If you did not complete this on your own, don't sweat it, closures are tough and take practice to get comfortable with.